openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go

```diff
@@ -136,7 +136,12 @@ func (o *DiscoverEtcdInitialClusterOptions) Run() error {
 	}
 
 	client, err := o.getClient()
-	if err != nil {
+	if err != nil && memberDirExists {
+		// we weren't able to get client and need to return based previous memberDir so we can restart.  This is the off and on again flow.
```

Comment:
If for some reason we get a transient error and the member is invalid, 
1. the best case is pod will crashloop and we try again next time error is not nil and we have a decision on what to do. 
2. the worst would-be etcd starts and spams the logs with cluster-id mismatch and does not restart, forcing an admin to manually remove the member.

I am trying to think of scenarios to validate/invalidate this, needs more thought IMHO.

/cc @hexfusion

Reply:
We hit this condition only if member-dir is present. member-dir is present only in two cases:
1. **Lights-out case**: then we should go ahead and return
2. Recovery case: we are supposed to archive by detecting the unstarted state. But in this case, it will go ahead and start.